icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -255,6 +258,76 @@ class SimpleUnitIdentifiersSink : public icu::ResourceSink {
     int32_t outIndex;
 };
 
+class UnitAliasesSink : public icu::ResourceSink {
+  public:
+    /**
+     * Constructor.
+     * @param unitAliases The output vector of unit alias identifiers (CharString).
+     * @param unitReplacements The output vector of replacements for the unit aliases (CharString).
+     */
+    explicit UnitAliasesSink(MaybeStackVector<CharString> &unitAliases,
+                             MaybeStackVector<CharString> &unitReplacements)
+        : unitAliases(unitAliases), unitReplacements(unitReplacements) {}
+
+    /**
+     * Adds the table keys (unit aliases) found in value to the unitAliases output vector and the
+     * replacements for the unit aliases to the unitReplacements output vector.
+     * @param key The key of the resource passed to `value`: the second
+     *     parameter of the ures_getAllItemsWithFallback() call.
+     * @param value Should be a ResourceTable value, if
+     *     ures_getAllItemsWithFallback() was called correctly for this sink.
+     * @param noFallback Ignored.
+     * @param status The standard ICU error code output parameter.
+     */
+    void put(const char * /*key*/, ResourceValue &value, UBool /*noFallback*/,
+             UErrorCode &status) override {
+        ResourceTable table = value.getTable(status);
+        if (U_FAILURE(status)) return;
+
+        // Collect the unit aliases from the table resource.
+        const char *unitAlias;
+        for (int32_t i = 0; table.getKeyAndValue(i, unitAlias, value); ++i) {
```

Comment:
You might want to consider using `getAllChildrenWithFallback()` instead of `getAllItemsWithFallback()`.  It'd let you avoid this loop and might help if instances of this resource ever use sideways inheritance.

Reply:
Great suggestion, done :)